check docker permissions without groups

xoebus opened the pull request:
I originally wanted this fix because the privilege escalation is not
required on MacOS but ended up tunnel-visioning on the implementation
for the Linux workstation I was working on. This caused two issues:

  1) It encouraged users to add the docker group to their user which has
     a number of security caveats[1] as reported by @jfmyers9.

  2) The fix doesn't even work on MacOS.

This new implementation checks if a user can use sudo-less docker
commands by actually trying to run one rather than trying to infer it
from the environment. This addresses the two points above.

[1]: https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface

xoebus commented:
I checked this patch on my Macbook which does not require `sudo` and my Linux workstation which does.

jfmyers9 commented:
Should we tackle https://github.com/cloudfoundry-incubator/bpm-release/blob/master/scripts/start-docker as part of this PR as well? That script is useful when developing locally.

xoebus commented:
If we're happy with this approach then I can apply this to all of the docker invocations and rebase it onto this.

xoebus commented:
Ok, I've applied the same changes to both files and extracted the function into a source-able file. I also moved the comment from the test-with-docker script about development workstations to describe a more apt location where `sudo` is used.
